core/locale/domain/date_time_test.go

```diff
@@ -0,0 +1,155 @@
+package domain
+
+import (
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDateTimeFormatter_SetDateTime(t *testing.T) {
+	formatter := &DateTimeFormatter{}
+	assert.Equal(t, &DateTimeFormatter{}, formatter)
+
+	now := getUTCNow()
+	loc, err := time.LoadLocation("Europe/Berlin")
+	assert.NoError(t, err)
+
+	formatter.SetDateTime(now, now.In(loc))
+	assert.Equal(t, &DateTimeFormatter{
+		dateTime:      now,
+		localDateTime: now.In(loc),
+	}, formatter)
+}
+
+func TestDateTimeFormatter_SetLocation(t *testing.T) {
+	now := getUTCNow()
+
+	formatter := &DateTimeFormatter{
+		logger:   flamingo.NullLogger{},
+		dateTime: now,
+	}
+
+	assert.Errorf(t, formatter.SetLocation("wrong"), ErrInvalidLocation)
```

Comment:
Instead usage of ErrInvalidLocation the message should be "Expected Error on invalid location" - or better use assert.EqualError ?